### DISCUSSION TITLE
Detector's detect() method returns nothing

### DISCUSSION THREAD

**Discussion by alexyuisme** [AlexeyAB/darknet#7760] [Q&A]
Hello, 

I got yolo_cpp_dll compiled and work but the Detector failed to detect objects from an image:

std::string file_name = "C:\person.jpg";
std::vector<bbox_t> boxes;
boxes = detector->detect(file_name); // ---> boxes.size() is 0.

But It works if I run the command line to test the same image.

.\darknet detect cfg/yolov4.cfg yolov4.weights data/person.jpg

Any possible reasons?

Thanks,
Alex

**Comment by alexyuisme** [comment]
Tracing down the problem from the source, the issue is l.output[obj_index]  == nan for all three yolo layers. 

int yolo_num_detections(layer l, float thresh)
{
    int i, n;
    int count = 0;
    for(n = 0; n < l.n; ++n){
        for (i = 0; i < l.w*l.h; ++i) {
            int obj_index  = entry_index(l, 0, n*l.w*l.h + i, 4);
            if(l.output[obj_index] > thresh){ // -----> l.output[obj_index] == nan !!
                ++count;
            }
        }
    }
    return count;
}

**Comment by alexyuisme** [comment]
fix is found: one needs to remove CUDNN_HALF pre-compile flag before compiling yolo_cpp_dll.

**Comment by CemEntok** [comment]
Hey @alexyuisme 
- Why do you need compiling yolo_cpp_dll?
- Did you also try this and make this worked?
`./darknet detector test obj.data yolov4-obj.cfg yolov4-obj_best.weights -ext_output -dont_show < test.txt > result.txt`
